apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/AbstractTracerContext.java

```diff
@@ -115,4 +116,10 @@
      */
     void asyncStop(AsyncSpan span);
 
+    /**
+     * Get current context profiler
+     * @return null if not profiling
+     */
+    ThreadProfiler profiler();
```

Comment:
We should not open this heavy object for this. Right now, only true/false should be opened.